elasticsearch/init.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash -e
+#
+# Copyright 2018 Red Hat, Inc. and/or its affiliates
+# and other contributors as indicated by the @author tags.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source "logging"
+
+LOG_FILE=${LOG_FILE:-elasticsearch_connect_log.txt}
+
+wait_for_port_open() {
+    rm -f $LOG_FILE
+    # test for ES to be up first and that our SG index has been created
+    info "Checking if Elasticsearch is ready on $ES_REST_BASEURL"
+    while ! response_code=$(es_util --query=/ \
+        ${DEBUG:+-v} -s \
+        --request HEAD --head \
+        --max-time $max_time \
+        -o $LOG_FILE -w '%{response_code}') || test $response_code != "200"
+    do
+        sleep $RETRY_INTERVAL
+        (( retry -= 1 )) || :
+        if (( retry == 0 )) ; then
+            timeouted=true
+            break
+        fi
+    done
+
+    if [ $timeouted = true ] ; then
+        error "Timed out waiting for Elasticsearch to be ready"
+    else
+        rm -f $LOG_FILE
+        info Elasticsearch is ready and listening
+        return 0
+    fi
+    cat $LOG_FILE
+    rm -f $LOG_FILE
+    exit 1
+}
+
+pushd "${ES_HOME}/init"
+  files=$(ls ${ES_HOME}/init | sort)
+  for init_file in $(ls ${ES_HOME}/init | sort); do 
+      set +e
+      if [ -f $init_file ] ; then
+        $(./init-file)
```

Comment:
what about `./$init_file`?
Also, maybe you want to execute all init scripts, and return an error if any of them returns an error?  If so, then use something like this
```
    rc=0
    ....
   ./$init_file || rc=1
   ....
   exit $rc
```
then you can leave `set -e` on all the time - the `||` will "catch" the error